modules/charon-core/src/main/java/org/wso2/charon3/core/encoder/JSONEncoder.java

```diff
@@ -116,19 +117,39 @@ public JSONObject getSCIMObjectAsJSONObject(SCIMObject scimObject) throws Charon
             //encode schemas
             this.encodeArrayOfValues(SCIMConstants.CommonSchemaConstants.SCHEMAS,
                     (scimObject.getSchemaList()).toArray(), rootObject);
-            //encode attribute list
+            //relate attributes to schema
+            Map<String, List<Attribute>> schemaAttributeMap = new HashMap<>();
+            scimObject.getSchemaList().stream().forEach(schema -> schemaAttributeMap.put(schema, new ArrayList<>()));
             Map<String, Attribute> attributes = scimObject.getAttributeList();
             if (attributes != null && !attributes.isEmpty()) {
                 for (Attribute attribute : attributes.values()) {
+                    Optional<String> schema = scimObject.getSchemaList().stream()
+                        .filter(attribute.getURI()::startsWith).findFirst();
+                    if (!schema.isPresent()) {
+                        throw new CharonException("Schema not present for attribute " + attribute.getURI());
```

Comment:
Is there any specific reason to throw an exception if the attribute.getURI() is not present in the schema list?

Reply:
Each attribute needs to be related to the schema where it is defined. This is because the schema URI is printed out in the JSON string representation -  which is the subject of this PR.

Do you have any suggestions on how to handle attributes that do not belong to any schema? 
When decoding those attributes, the JSONDecoder would omit the attributes, because they are unknown. Do you think just ignoring those attributes is better than throwing the exception?

Reply:
attribute.getURI()::startsWith will provoke NPE in JSONEncoder with the current code of, for example, GroupResourceManager#listWithPOST because we will have "totalResults" attribute there without URI...

Reply:
@denis111:  I am sorry about this issue! 

I created a new PR #188 which fixes the NPE. Please have a look.

@IsuraD: There are no tests for this project yet. Or am I missing anything? What about introducing tests for each new PR? This would have avoided the NPE.